packages/environment/chain.js

```diff
@@ -35,6 +35,10 @@ try {
 
 options.time = options.time ? new Date(options.time) : new Date();
 
+options.vmErrorsOnRPCResponse = "vmErrorsOnRPCResponse" in options ? options.vmErrorsOnRPCResponse : true;
+
+options.legacyInstamine = "legacyInstamine" in options ? options.legacyInstamine : true;
+
```

Comment:
By making `vmErrorsOnRPCResponse: true` and `legacyInstamine: true` ganache's error reporting behavior and mining behavior is mostly restored to the way it worked in v2.